### PR TITLE
Add definition to log.Print() func

### PR DESCRIPTION
**Pull request by SupriyaKasten**
## Change Overview

Function definition for log.Print() func

## Pull request type

Please check the type of change your PR introduces:
- [ ] Work in Progress
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Trivial/Minor
- [ ] Bugfix
- [x] Feature
- [ ] Documentation

## Test Plan

- [ ] Manual
- [ ] Unit test
- [ ] E2E

### CODE REVIEW

File: pkg/log/log.go [hunk 1]
```diff
@@ -46,23 +48,62 @@ func WithContext(ctx context.Context) {
 	Info().WithContext(ctx)
 }
 
-func (l *logger) Print(msg string) {
+func (l *logger) Print(msg string, fields field.Fields) {
+	logFields := make(logrus.Fields)
+	if l.ctx != nil {
+		ctxFields := field.FromContext(l.ctx)
```
Comment (tdmanv) [inline pkg/log/log.go]: does this handle nil? Can we make it handle nil?

File: pkg/log/log.go [hunk 2]
```diff
@@ -46,23 +48,62 @@ func WithContext(ctx context.Context) {
 	Info().WithContext(ctx)
 }
 
-func (l *logger) Print(msg string) {
+func (l *logger) Print(msg string, fields field.Fields) {
```
Comment (tdmanv) [inline pkg/log/log.go]: Please decouple accepting fields as a param and extracting the fields from the context.

Comment (julio-lopez) [reply pkg/log/log.go]: For now, let's have the `Print` signature be:
```suggestion
func (l *logger) Print(msg string) {
```
The `fields` argument is not immediately needed, right? They will be extracted from the context, no?
We can easily extend the `Print` function later to accept other arguments. However, we have not gone through the exercise of evaluating the tradeoffs of different alternatives.
We decided to use `WithContext()` and `WithError()` for now as a mean to achieve the immediate desired functionality.

File: pkg/log/log.go [hunk 3]
```diff
@@ -46,23 +48,50 @@ func WithContext(ctx context.Context) {
 	Info().WithContext(ctx)
 }
 
+func WithError(err error) {
+	Error().WithError(err)
+}
+
 func (l *logger) Print(msg string) {
+	logFields := make(logrus.Fields)
+	if ctxFields := field.FromContext(l.ctx); ctxFields != nil {
```
Comment (tdmanv) [inline pkg/log/log.go]:
```suggestion
	if ctxFields := field.FromContext(l.ctx); len(ctxFields) != 0 {
```

Comment (SupriyaKasten) [reply pkg/log/log.go]: @tdmanv ctxFields is of type field.Fieldser interface. I think `len(ctxFields)` will fail.

Comment (tdmanv) [reply pkg/log/log.go]: you're right! please ignore this comment

File: pkg/log/log.go [hunk 4]
```diff
@@ -46,23 +48,50 @@ func WithContext(ctx context.Context) {
...
+		l.entry.Error(msg)
 	case DebugLevel:
-		logrus.Debug(msg)
+		l.entry.Debug(msg)
 	}
 }
 
 func (l *logger) WithContext(ctx context.Context) Logger {
 	l.ctx = ctx
+	if l.entry != nil {
```
Comment (julio-lopez) [inline pkg/log/log.go]: Later we can figure out an idiomatic way of avoiding having to always check whether entry is `nil` or not.
And in particular, it is desirable to make "logger" immutable. That's mostly implementation and would be compatible with the API currently exposed.

**Pull request merged**